src/wayland/input_method/input_method_handle.rs

```diff
@@ -69,39 +70,15 @@ impl InputMethodHandle {
         keyboard.grab.is_some()
     }
 
-    /// Convenience function to draw surfaces
-    pub fn with_surface<F>(&self, mut f: F)
-    where
-        F: FnMut(&WlSurface),
-    {
-        let inner = self.inner.lock().unwrap();
-        let popup = inner.popup.inner.lock().unwrap();
-        if popup.surface_role.is_some() {
-            if let Some(surface) = &popup.surface {
-                if surface.alive() {
-                    f(surface);
-                }
-            }
-        }
+    pub(crate) fn set_text_input_rectangle(&self, x: i32, y: i32, width: i32, height: i32) {
```

Comment:
Could we at some point in the chain (this function or `set_rectangle` not sure where the best point is) convert this into a `Rectangle` type please?

Reply:
I did in the next point of the chain, but I have now moved it forward, also trimmed some code on the way so all is good :rocket: